services/order.py

```
@@ -0,0 +1,50 @@
+from datetime import datetime
+
+from django.contrib.auth import get_user_model
+from django.db import transaction
+from django.db.models import QuerySet
+
+from db.models import Ticket, Order, MovieSession
+
+
+@transaction.atomic
+def create_order(
+        tickets: list[dict],
+        username: str,
+        date: str = None
+) -> Order:
+    order = Order(
+        user=get_user_model().objects.get(username=username),
+    )
+    order.save()
+
+    if date:
+        order.created_at = datetime.strptime(date, "%Y-%m-%d %H:%M")
+        order.save()
+
+    movie_session = MovieSession.objects.get(pk=tickets[0]["movie_session"])
+
+    tickets = Ticket.objects.bulk_create(
+        [
+            Ticket(
+                row=ticket_dict["row"],
+                seat=ticket_dict["seat"],
+                movie_session=movie_session,
+                order=order,
+            )
```

What if ticket have no data and you got KeyError? you can replace it  for  .get()

---

If the data in the ticket dictionary is invalid, we should throw an error, because if we don't, data consistency will be broken. Throwing an error when the dictionary is invalid, combined with the transaction mechanism, ensures that the order will only be created if valid ticket data has been passed.